hack/util.sh

```diff
@@ -460,17 +458,18 @@ function kill_all_processes()
 {
 	local sudo="${USE_SUDO:+sudo}"
 
-	pids=($(jobs -pr))
-	for i in ${pids[@]-}; do
-		pgrep -P "${i}" | xargs $sudo kill &> /dev/null
-		$sudo kill ${i} &> /dev/null
+	for parent in $(jobs -pr); do
+		for child in $(pgrep -P "${parent}"); do
+			"${sudo}" kill "${child}" &> /dev/null
```

Comment:
@Miciah is this really the best approach here? I remember when I was doing my refactors (that have yet to land ... sometime soon!) I found that sometimes killing the child will corrupt the parent, and sometimes the parent has a 'keepalive' type of relationship with it's children.

Should we instead kill the parent first, and then later try killing the children if they're still alive?

Should we be explicitly using `kill -SIGTERM` here, and a follow-up `kill -SIGKILL` after some timeout? I believe I do that in my refactor so maybe most of these changes should just wait until that.